### DISCUSSION TITLE
searchparams instantly set to null when page.tsx is in a parentheses folder

### DISCUSSION THREAD

**Discussion by Jip-van-Sommeren** [vercel/next.js#60638] [Help]
### Summary

I want to acces my set searchparams in a page in parentheses folder (e.g. (some_folder)), however the searchparams are initially set correctly, however they almost instantly get set to null, this does not happen when they are not in a folder of this type, is there any way to overcome this?

**Comment by nirvikpurkait** [comment]
Parathesis folders are used to make a route group, are you able to access the content in `page.tsx` of paranthesis folder?

**Reply by Jip-van-Sommeren** [reply]
When i log the searchParams, it logs the correct searchparams, but almost instantly after they get set to Null, this doesnt happen If it is not in the folder

**Reply by nirvikpurkait** [reply]
Can you provide your file and folder structure?

**Reply by nirvikpurkait** [reply]
And are you using client component?

**Comment by Jip-van-Sommeren** [comment]
Its in app/dashboard/(menu)/page.tsx If I change the folder name to just menu, it works as intended. If I remover the folder all together it works as intended. no “use client”

**Reply by nirvikpurkait** [reply]
Ok, so see the paranthesis folder are used to make route group. If you make `page.tsx` inside `/dashbord/(menu)` the content does not get serverd for your request, only the content of `/dashboard/page.tsx` or `/dashboard/(menu)/another-route/page.tsx` gets serverd for their corresponding request. As the `/dashboard/(menu)/page.tsx` does not gets served, you don't even need to access the `serchParams`.

Now here is a catch, if you do not have `page.tsx` inside `/dashboard`, then you can accss `searchParams` in `page.tsx` of `/dashboard/(menu)`

If you feel stuck you can read more about route group [here](https://nextjs.org/docs/app/building-your-application/routing/route-groups)
